py symbolic: DeprecationWarning not raised by `e_xv == e_yv`?

Example failure: https://drake-cdash.csail.mit.edu/testDetails.php?test=25813237&build=1103482
Snippet:
```
FAIL: test_relational_operators_nonzero (symbolic_test.TestSymbolicExpression)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./bindings/pydrake/test/symbolic_test.py", line 528, in test_relational_operators_nonzero
    value = (e_xv == e_yv)
AssertionError: DeprecationWarning not raised
```

My guess is that this machine has a later version of NumPy in which the warning message has changed from what we're filtering for:
https://github.com/RobotLocomotion/drake/blob/14f5b3960/bindings/pydrake/common/deprecation.py#L175-L179

--- a/bindings/pydrake/common/deprecation.py
+++ b/bindings/pydrake/common/deprecation.py
@@ -165,10 +165,10 @@ def install_numpy_warning_filters(force=False):
         return
     _installed_numpy_warning_filters = True
     # Warnings specific to comparison with `dtype=object` should be raised to
-    # errors (#8315, #8491). Without them, NumPy will return effectively
-    # garbage values (e.g. comparison based on object ID): either a scalar bool
-    # or an array of bools (based on what objects are present and the NumPy
-    # version).
+    # errors (#8315, #8491). Without them, NumPy will swallow the errors and
+    # make a DeprecationWarning, while returning effectively garbage values
+    # (e.g. comparison based on object ID): either a scalar bool or an array of
+    # bools (based on what objects are present and the NumPy version).
     # N.B. Using a `module=` regex filter does not work, as the warning is
     # raised from C code, and thus inherits the calling module, which may not
     # be "numpy\..*" (numpy/numpy#10861).
@@ -177,6 +177,10 @@ def install_numpy_warning_filters(force=False):
     warnings.filterwarnings(
         "error", category=DeprecationWarning,
         message="elementwise == comparison failed")
+    # Error changed in 1.16.0
+    warnings.filterwarnings(
+        "error", category=DeprecationWarning,
+        message="elementwise comparison failed")
 
 
 warnings.simplefilter('once', DrakeDeprecationWarning)

--- a/bindings/pydrake/test/symbolic_test.py
+++ b/bindings/pydrake/test/symbolic_test.py
@@ -11,6 +11,7 @@ import pydrake.symbolic as sym
 import pydrake.common
 from pydrake.test.algebra_test_util import ScalarAlgebra, VectorizedAlgebra
 from pydrake.util.containers import EqualToDict
+from pydrake.util.deprecation import install_numpy_warning_filters
 
 # TODO(eric.cousineau): Replace usages of `sym` math functions with the
 # overloads from `pydrake.math`.
@@ -306,6 +307,14 @@ class TestSymbolicVariables(SymbolicTestCase):
 
 
 class TestSymbolicExpression(SymbolicTestCase):
+    def setUp(self):
+        SymbolicTestCase.setUp(self)
+        # For some reason, something in how `unittest` tries to scope warnings
+        # causes the previous filters to be lost. Re-install here.
+        # TODO(eric.cousineau): This used to be necessary for PY3-only, but
+        # with NumPy 1.16, it became PY2 too. Figure out why.
+        install_numpy_warning_filters(force=True)
+
     def _check_scalar(self, actual, expected):
         self.assertIsInstance(actual, sym.Expression)
         # Chain conversion to ensure equivalent treatment.
@@ -511,18 +520,6 @@ class TestSymbolicExpression(SymbolicTestCase):
         e_yv = np.array([e_y, e_y])
         # N.B. In some versions of NumPy, `!=` for dtype=object implies ID
         # comparison (e.g. `is`).
-        # N.B. If `__nonzero__` throws, then NumPy swallows the error and
-        # produces a DeprecationWarning, in addition to effectively garbage
-        # values. For this reason, `pydrake.symbolic` will automatically
-        # promote these warnings to errors.
-        if six.PY3:
-            # For some reason, something in how `unittest` tries to scope
-            # warnings causes the previous filters to be lost. Re-install
-            # here.
-            # TODO(eric.cousineau): Figure out better hook for this, or better
-            # way to restore warning filters from when everything's imported.
-            from pydrake.util.deprecation import install_numpy_warning_filters
-            install_numpy_warning_filters(force=True)
         # - All false.
         with self.assertRaises(DeprecationWarning):
             value = (e_xv == e_yv)

--- a/common/test_utilities/drake_py_unittest_main.py
+++ b/common/test_utilities/drake_py_unittest_main.py
@@ -100,9 +100,13 @@ if __name__ == '__main__':
              "client (non-batch), output may be mixed, so piping makes " +
              "the output more readable.")
     parser.add_argument(
-        "--drake_deprecation_action", type=str, default="error",
-        help="Action for Drake deprecation warnings. See " +
+        "--deprecation_action", type=str, default="once",
+        help="Action for any deprecation warnings. See " +
              "`warnings.simplefilter()`.")
+    parser.add_argument(
+        "--drake_deprecation_action", type=str, default="error",
+        help="Action for Drake deprecation warnings. Applied after " +
+             "--deprecation_action.")
     args, remaining = parser.parse_known_args()
     sys.argv = sys.argv[:1] + remaining
 
@@ -119,6 +123,9 @@ if __name__ == '__main__':
         sys.stdout.flush()
         sys.stdout = sys.stderr
 
+    # Ensure deprecation warnings are always shown at least once.
+    warnings.simplefilter(args.deprecation_action, DeprecationWarning)
+    # Handle Drake-specific deprecations.
     if has_pydrake:
         warnings.simplefilter(
             args.drake_deprecation_action, DrakeDeprecationWarning)

